build/ris.js

```diff
@@ -0,0 +1,397 @@
+            res.on('data', (chunk) => {
+                responseBody += chunk;
+            });
+
+            res.on('end', () => {
+                resolve(JSON.parse(responseBody));
+            });
+        });
+        req.on('error', (err) => {
+            reject(err);
+        });
+        req.write(data)
+        req.end();
+    });
+}
+
+const doRisStationsRequest = async (latitude, longitude, radius, groupBy) => {
+    const query = {
+        latitude,
+        longitude,
+        radius,
+        groupBy,
+        onlyActive: false,
+        limit: LIMIT_RESULTS
+    }
+    const options = {
+        hostname: 'apis.deutschebahn.com',
+        port: 443,
+        path: '/db-api-marketplace/apis/ris-stations/v1/stop-places/by-position?' + stringify(query),
+        method: 'GET',
+        headers: {
+            'DB-Client-Id': process.env.DB_CLIENT_ID,
+            'DB-Api-Key': process.env.DB_API_KEY,
+        },
+    };
+    console.error(options.path);
+    return doRequest(options, '');
+}
+
+const doStadaRequest = async () => {
+    const options = {
+        hostname: 'apis.deutschebahn.com',
+        port: 443,
+        path: '/db-api-marketplace/apis/station-data/v2/stations',
+        method: 'GET',
+        headers: {
+            'DB-Client-Id': process.env.DB_CLIENT_ID,
+            'DB-Api-Key': process.env.DB_API_KEY,
+            'Accept': 'application/json'
+        },
+    };
+    console.error(options.path);
+    return doRequest(options, '');
+}
+
+const geodist = (lat1, lon1, lat2, lon2) => {
+    const l1 = lat1 * Math.PI / 180, l2 = lat2 * Math.PI / 180, delta = (lon2 - lon1) * Math.PI / 180, R = 6371e3;
+    return Math.acos(Math.sin(l1) * Math.sin(l2) + Math.cos(l1) * Math.cos(l2) * Math.cos(delta)) * R;
+}
+
+const discoverRisStations = async (bbox, depth, num_requests) => {
+    if (depth > MAX_DEPTH) {
+        throw new Error('Emerg: Max depth reached');
+    }
+    await sleep(5000 * Math.random());
+    const halfX = (bbox.east - bbox.west) / 2;
+    const halfY = (bbox.north - bbox.south) / 2;
+    const center = {
+        lat: halfY + bbox.south,
+        lon: halfX + bbox.west
+    }
+    const radius = Math.round(geodist(center.lat, center.lon, bbox.south, bbox.east) * PADDING_RATIO);
+    const fileName = TMP_PATH + depth + '_' + Object.values(bbox).join('_') + '_single.json';
+    const fileExists = await exists(fileName);
+
+    const dx = geodist(center.lat, center.lon, center.lat, bbox.east), dy = geodist(center.lat, center.lon, bbox.south, center.lon);
+    console.error('Descending to depth', depth, 'with bbox', bbox, 'and radius', radius, 'and center', center.lat, center.lon, 'and ratio', dx, dy, dx / dy);
+    let single;
+    if (!fileExists) {
+        console.error(fileName);
+        single = await doRisStationsRequest(center.lat, center.lon, radius, 'NONE');
+        num_requests += 1;
+        await fs.promises.writeFile(fileName, JSON.stringify(single), 'utf8', err => {
```

Comment:
`fs.promises.writeFile()` (or `fs/promises`' `writeFile()`, which is equivalent) doesn't take a callback as the last parameter.